_posts/2022-08-05-hacking-for-cruise-control-ui.md

```diff
@@ -0,0 +1,305 @@
+Regardless of the reason, we want to provide some basic instructions for getting you started!
+
+## Prerequisites
+
+For starters we will need a Strimzi-managed Kafka cluster with Cruise Control enabled.
+
+![](/assets/images/posts/2022-10-31-hacking-for-cruise-control-ui-1.png)
+
+Since Strimzi does not support creating custom Cruise Control API user roles we will need to disable HTTP basic authentication settings.
+
+We can do so by configuring Cruise Control within a `Kafka` resource like this:
+
+``` 
+# kafka-cruise-control.yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  cruiseControl:
+    config:
+      webserver.security.enable: false
+```
+
+After configuring our Strimzi-managed Kafka cluster, we can deploy and connect the [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) in one of two ways:
+
+* METHOD ONE: Inside a dedicated Cruise Control pod
+* METHOD TWO: Outside a dedicated Cruise Control pod
+
+## METHOD ONE: Inside a dedicated Cruise Control pod 
+
+With our Strimzi-managed Kafka cluster deployed, we can create and deploy a custom Cruise Control image which includes the [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) application inside of it.
+
+![](/assets/images/posts/2022-10-31-hacking-for-cruise-control-ui-1.png)
+
+
+### Inside the custom Cruise Control container
+
+Here the [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) will run alongside the Cruise Control application in the same container making its requests directly to the Cruise Control REST API.
+
+![](/assets/images/posts/2022-10-31-hacking-for-cruise-control-ui-3.png)
+
+### Building the Cruise Control UI container
+
+We can base our custom Cruise Control image on the original Strimzi Kafka image and install the [Cruise Control UI](https://github.com/linkedin/cruise-control-ui) bits on top:
+
+```bash
+#Dockerfile
+FROM quay.io/strimzi/kafka:latest-kafka-3.2.3
+
+ENV CC_UI_VERSION=0.4.0
+ENV CC_UI_HOME=./cruise-control-ui/dist/
+
+USER root
+
+RUN mkdir -p ${CC_UI_HOME}
+
+RUN curl -LO https://github.com/linkedin/cruise-control-ui/releases/download/v${CC_UI_VERSION}/cruise-control-ui-${CC_UI_VERSION}.tar.gz; \
+    tar xvfz cruise-control-ui-${CC_UI_VERSION}.tar.gz -C ${CC_UI_HOME} --strip-components=2; \
+    rm -f cruise-control-ui-${CC_UI_VERSION}.tar.gz*; \
+    echo "dev,dev,/kafkacruisecontrol/" > "${CC_UI_HOME}"static/config.csv;
+
+USER 1001
+```
+
+Then building and pushing the image to a container registry:
+```
+# When building, tag the image with the name of a container registry
+# that is accessible by the Kubernetes cluster.
+docker build . -t <registry>/<cruise-control-with-ui-image-name>
+
+# Push the image to that container registry
+docker push <registry>/<cruise-control-with-ui-image-name>
+```
+
+### Deploying the custom Cruise Control pod
+
+To deploy our custom Cruise Control pod we must edit our Strimzi Cluster Operator `Deployment`:
```

Comment:
Not sure if editing the CO deployment is the best way.
If it's already running, it means that you need to restart to get a new CC image to be used.
If CC is already running, it would mean two rolls (CO + CC).
In general, I prefer to leave the default images as they are and using the `image` field, so in this case the `cruiseControl.image` field in the `Kafka` CR. It will just let the CO to detect the change and roll a CC if already running but then using the new image.
It's just my opinion, it's worth taking others ... @scholzj @tombentley @mimaison wdyt?